nyckel.py

```diff
@@ -135,6 +135,72 @@ def invoke(access_token, function_id, ablationSize, dataset):
                 print(f"Total: {total}")
 
 
+def parallel_invoke(access_token, function_id, ablationSize, dataset):
+
+    url = f"https://www.nyckel.com/v1/functions/{function_id}/invoke"
+
+    headers = {"Authorization": f"Bearer {access_token}"}
+
+    with open(f"data/{dataset}/{dataset}_test_nyckel.csv") as csvfile:
+        reader = csv.reader(csvfile)
+        # get the class labels from classes.txt
+        count = 0
+        image_array = []
+        for row in reader:
+            print(row[0])
+            f = open(f"data/{dataset}/test/{row[1]}/{row[0]}", "rb")
+            image_array.append(f)
+            count += 1
+            if count == 10:
+                break
+  
+    def _parallel_invoke(image_array):
+        for f in image_array:
+            response = requests.post(url, headers=headers, files={"data": f})
+            print(response.json())
+
+    start = time.time()
+    Parallel(n_jobs=10, prefer="threads")(
+        delayed(_parallel_invoke)(image_array)
+        for f in image_array
+    )
+    end = time.time()
+    print(f"Time to 10 invokes {dataset}-{ablationSize}: {end - start}")
+
+
+def threaded_invoke(access_token, function_id, ablationSize, dataset):
+
+    start = time.time()
+    url = f"https://www.nyckel.com/v1/functions/{function_id}/invoke"
+
+    headers = {"Authorization": f"Bearer {access_token}"}
+
+    with open(f"data/{dataset}/{dataset}_test_nyckel.csv") as csvfile:
+        reader = csv.reader(csvfile)
+        # get the class labels from classes.txt
+        count = 0
+        image_array = []
+        for row in reader:
+            print(row[0])
+            f = open(f"data/{dataset}/test/{row[1]}/{row[0]}", "rb")
+            image_array.append(f)
+            count += 1
+            if count == 10:
```

Comment:
So you only call the API 10 times using 10 threads. I'd suggest calling it like 1000 times using 10 threads. That should smooth out throughput measurements.

Reply:
I second ^

Reply:
Another thing to keep in mind - only count successful (http status code 200) responses towards throughput.